guide/self_hosted_registry.md

```diff
@@ -0,0 +1,159 @@
+# Using a self-hosted registry with OpenFaaS
+
+If you're using OpenFaaS on single host, then you don't need to push your images to a registry. They will just be used from the local Docker library.
+
+If you are using a remote server or a multi-node cluster then you can push your function's image to a registry or the Docker Hub.
+
+This describes how to use OpenFaaS in a swarm with your own local registry for hosting function images.
+
+## Set up a swarm
+
+For this example lets presume you want to create a swarm of 3 nodes. Use node1 as the manager.
+
+```
+$ docker swarm init --advertise-addr $(hostname -i)
+```
+
+Now in node2 and node3 paste the output from the above command.
+```
+$ docker swarm join --token ...
+```
+
+## Install OpenFaaS
+```
+$ git clone https://github.com/openfaas/faas && \
+  cd faas && \
+  ./deploy_stack.sh
+```
+
+## Add a registry to the swarm
+
+```
+docker service create -d -p 5000:5000 \
+  --network func_functions --detach=false \
+  --constraint 'node.role == manager' \
+  --name registry registry:2
+```
+
+Here we are using a basic local registry. You can deploy it elsewhere and use volumes depending on your persistence needs. If you would like to [use authentication with your registry, this guide may be helpful](https://github.com/openfaas/faas/blob/master/docs/managing-images.md#deploy-your-own-private-docker-registry).
```

Comment:
This linked document needs updating for terminology. Would you be able to cover that @jonocodes ?

Reply:
The document looks good. What terminology am I looking for? Perhaps these docs should be merged into 1 somehow?